feat: Tradingview setting

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a developer setting to switch between production and test versions of TradingView with dynamic labels in settings.
	- Added a toggle for enabling the TradingView test domain in the settings interface.
- **Style**
	- Updated the auto jump switch to a smaller size for a more consistent UI.
- **Bug Fixes**
	- Simplified dialog handling for the WebEmbed configuration to improve code clarity.
- **Chores**
	- Disabled caching in the native web view component for improved loading behavior.
	- Enhanced web view behavior to accommodate the TradingView test domain setting.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: add TradingView settings to DevSettings and update useTradingViewProps for test domain support

- Introduced new settings for showing TradingView and using a test domain in the DevSettings section.
- Updated useTradingViewProps to conditionally use the test domain based on the new settings.
- Enhanced the DevSettings UI to include toggles for these new features.

limichange committed:
refactor: remove showTradingView setting from DevSettings and UI

- Eliminated the showTradingView setting from the DevSettings interface and the corresponding UI component in the DevSettingsSection.
- Streamlined the settings section by removing the toggle for showing TradingView.

limichange committed:
refactor: improve formatting in DevSettingsSection and devSettings interface

- Enhanced code readability by adjusting the formatting of string literals in the DevSettingsSection component.
- Updated the devSettings interface to improve the alignment of the showDevOverlayWindow property for better clarity.

limichange committed:
feat: add small size option for switch in AutoJumpSetting component

- Introduced ESwitchSize.small to the Switch component in the AutoJumpSetting for improved UI consistency.
- Enhanced the user experience by providing a smaller switch size option.

limichange committed:
refactor: update titles and subtitles in DevSettingsSection for clarity

- Changed the title of the TradingView test domain setting to "Use TradingView Test" for brevity.
- Updated the subtitle to display "新版 TradingView" and "旧版 TradingView" based on the test domain setting state, enhancing user understanding.

limichange committed:
refactor: improve conditional styling in useTradingViewProps

- Enhanced the conditional rendering of styles in the useTradingViewProps function for better readability.
- Adjusted the formatting of the string literals to streamline the code structure.

limichange committed:
Merge branch 'x' into tradingview-setting

originalix committed:
Merge branch 'x' into tradingview-setting

limichange committed:
refactor: streamline WebView component and project settings

- Updated the WebView component to include a conditional check for the TradingView test domain, enhancing flexibility in loading requests.
- Adjusted the project settings in Xcode to simplify the OTHER_LDFLAGS configuration, improving build clarity.
- Disabled caching in the NativeWebView for better performance consistency.

limichange committed:
refactor: update OTHER_LDFLAGS configuration in Xcode project

- Modified the OTHER_LDFLAGS setting in the Xcode project to use an array format, improving clarity and maintainability.
- Ensured consistent formatting across multiple build configurations.

limichange committed:
fix: add missing newline at end of Xcode project file

- Added a newline at the end of the project.pbxproj file to comply with standard file formatting practices, ensuring better compatibility with version control systems.

sidmorizon commented on packages/kit/src/components/TradingView/WebView.native.tsx:
```diff
@@ -21,6 +22,8 @@ export function WebView({
   onLoadEnd: () => void;
 }) {
+  const [devSettings] = useDevSettingsPersistAtom();
+  const isTest = devSettings.settings?.useTradingViewTestDomain;
```
devSettings.settings?.isEnabled && devSettings.settings?.useTradingViewTestDomain;

originalix committed:
Merge branch 'x' into tradingview-setting

originalix committed:
Merge branch 'x' into tradingview-setting

limichange committed:
refactor: enhance TradingView test domain checks in useTradingViewProps and WebView

- Updated the conditional logic for the TradingView test domain in both useTradingViewProps and WebView components to ensure it only activates when devSettings is enabled.
- Improved code clarity by removing redundant references to the test domain setting in useTradingViewProps.

limichange committed:
Merge branch 'x' into tradingview-setting

limichange committed:
Merge branch 'x' into tradingview-setting

huhuanming committed:
Merge branch 'x' into tradingview-setting
